File: core/common/src/main/java/com/shadov/codehellven/common/model/Difficulty.java
package com.shadov.codehellven.common.model;

import com.fasterxml.jackson.annotation.JsonCreator;

public enum Difficulty {
	STARTING(1), EASY(3), MEDIUM(5), HARD(10), GOD(50);

	private int reputation;

	Difficulty(int reputation) {
		this.reputation = reputation;
	}

	public int reputation() {
		return reputation;
	}

	@JsonCreator
	public static Difficulty fromString(String key) {
		for (Difficulty type : Difficulty.values()) {
			if (type.name().equalsIgnoreCase(key))
				return type;
		}
		return null;
	}
}

File: core/common/src/main/kotlin/com/shadov/codehellven/common/model/Difficulty.kt
package com.shadov.codehellven.common.model

import com.fasterxml.jackson.annotation.JsonCreator

enum class Difficulty constructor(private val reputation: Int) {
    STARTING(1), EASY(3), MEDIUM(5), HARD(10), GOD(50);

    fun reputation(): Int {
        return reputation
    }

    companion object {
        @JsonCreator
        fun fromString(key: String): Difficulty {
            for (type in Difficulty.values()) {
                if (type.name.equals(key, ignoreCase = true))
                    return type
            }
            throw IllegalArgumentException("Difficulty with key $key does not exist")
        }
    }
}
